src/hooks/video/agora/index.ts

```diff
@@ -0,0 +1,233 @@
+import { useState, useEffect, useCallback } from "react";
+import AgoraRTC, {
+  IAgoraRTCRemoteUser,
+  ILocalAudioTrack,
+  ILocalVideoTrack,
+} from "agora-rtc-sdk-ng";
+
+import { AGORA_APP_ID, AGORA_CHANNEL, AGORA_TOKEN } from "secrets";
+
+import { useShowHide } from "hooks/useShowHide";
+import { useUser } from "hooks/useUser";
+import { useVenueId } from "hooks/useVenueId";
+
+import { updateTalkShowStudioExperience } from "api/profile";
+
+import { ReactHook } from "types/utility";
+import {
+  UseAgoraCameraProps,
+  UseAgoraCameraReturn,
+  UseAgoraRemotesProps,
+  UseAgoraRemotesReturn,
+  UseAgoraScreenShareProps,
+  UseAgoraScreenShareReturn,
+} from "types/agora";
+
+export const useAgoraRemotes: ReactHook<
+  UseAgoraRemotesProps,
+  UseAgoraRemotesReturn
+> = ({ client }) => {
+  const [remoteUsers, setRemoteUsers] = useState<IAgoraRTCRemoteUser[]>([]);
+
+  const updateRemoteUsers = useCallback(() => {
+    if (!client) return;
+
+    setRemoteUsers(() => Array.from(client.remoteUsers));
+  }, [client]);
+
+  const handleUserPublished = useCallback(
+    async (user: IAgoraRTCRemoteUser, mediaType: "audio" | "video") => {
+      if (!client) return;
+
+      await client.subscribe(user, mediaType);
+      updateRemoteUsers();
+    },
+    [client, updateRemoteUsers]
+  );
+
+  useEffect(() => {
+    if (!client) return;
+
+    updateRemoteUsers();
+
+    client.on("user-published", handleUserPublished);
+    client.on("user-unpublished", updateRemoteUsers);
+    client.on("user-joined", updateRemoteUsers);
+    client.on("user-left", updateRemoteUsers);
+    client.join(AGORA_APP_ID || "", AGORA_CHANNEL || "", AGORA_TOKEN || null);
```

Comment:
Calling out usages of `AGORA_APP_ID`, `AGORA_CHANNEL`, and `AGORA_TOKEN` that are likely wrong and almost certainly will need refactoring.